eth/vm/forks/frontier/__init__.py

```diff
@@ -47,16 +49,6 @@ def make_frontier_receipt(base_header: BlockHeaderAPI,
         in computation.get_log_entries()
     ]
 
-    gas_remaining = computation.get_gas_remaining()
-    gas_refund = computation.get_gas_refund()
-    tx_gas_used = (
-        transaction.gas - gas_remaining
-    ) - min(
-        gas_refund,
-        (transaction.gas - gas_remaining) // 2,
-    )
-    gas_used = base_header.gas_used + tx_gas_used
```

Comment:
Interestingly, this seems to remove *all* usages of `transaction` _and_ `base_header`, so it seems best to remove them as parameters. I interpret that as a good sign, that the method got simpler.

Reply:
Oops, nice catch!